policy/modules/services/glusterfs.if

```diff
@@ -1,5 +1,89 @@
 ## <summary>Cluster File System binary, daemon and command line.</summary>
 
+########################################
+## <summary>
+##	Execute glusterd in the glusterd domain.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to transition.
+##	</summary>
+## </param>
+#
+interface(`glusterfs_domtrans',`
+	gen_require(`
+		type glusterd_t, glusterd_exec_t;
+	')
+
+	corecmd_search_bin($1)
+	domtrans_pattern($1, glusterd_exec_t, glusterd_t)
+')
+
+########################################
+## <summary>
+##	Execute glusterd in the glusterd domain, and
+##	allow the specified role the glusterd domain.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to transition.
+##	</summary>
+## </param>
+## <param name="role">
+##	<summary>
+##	Role allowed access.
+##	</summary>
+## </param>
+## <rolecap/>
+#
+interface(`glusterfs_run',`
```

Comment:
These interfaces need some renaming. From the names themselves, it's not clear when they provide access to the CLI program or the daemon. I think `glusterfs_[verb]` should be operating on glusterd, and `glusterfs_[verb]_cli` would operate on the CLI tool.

Reply:
These interfaces are all for the daemon, but I agree a rename would help to clarify.

Reply:
I missed that. I'm not clear on why mount.glusterfs would need to run glusterd. To start glusterd if it's not running?

Reply:
It seems like `mount.glusterfs` runs the Gluster CLI which in turn runs `glusterd` at some point. For what reason I'm not sure yet. Investigating...

Reply:
So it seems `mount.glusterfs` will execute `glusterfs` which, on my system, is just a symlink to the gluster daemon binary. In my testing, `glusterfs` was started with these arguments:
```
/usr/sbin/glusterfs --process-name fuse --volfile-server=<server> --volfile-id=/<volume> <mount-path>
```

So yes, the domtrans is needed. Without it, we get:
```
/sbin/mount.glusterfs: line 406: /usr/sbin/glusterfs: Permission denied
Mounting glusterfs on /mnt/temp failed.
```